pkg/http/api.go

```diff
@@ -827,8 +839,8 @@ func (a *api) onGetWorkflowHandler() fasthttp.RequestHandler {
 func (a *api) onTerminateWorkflowHandler() fasthttp.RequestHandler {
```

Comment:
Here too, `onTerminateWorfklowHandler()`, `onResumeWorkflowHandler()` and `onPauseWorkflowHandler()` are pretty much identical down to th signature of `InModifier`. Please consider merging them into a single method

Reply:
Example:

```go
type workflowActivityHandler = func(ctx context.Context, in *runtimev1pb.WorkflowActivityRequest) (*runtimev1pb.WorkflowActivityResponse, error)

// Used by TerminateWorkflowAlpha1, PauseWorkflowAlpha1, ResumeWorkflowAlpha1
func (a *api) onWorkflowActivityHandler(handler workflowActivityHandler) fasthttp.RequestHandler {
	return UniversalFastHTTPHandler(
		handler,
		UniversalFastHTTPHandlerOpts[*runtimev1pb.WorkflowActivityRequest, *runtimev1pb.WorkflowActivityResponse]{
			InModifier: func(reqCtx *fasthttp.RequestCtx, in *runtimev1pb.WorkflowActivityRequest) (*runtimev1pb.WorkflowActivityRequest, error) {
				in.InstanceId = reqCtx.UserValue(instanceID).(string)
				in.WorkflowComponent = reqCtx.UserValue(workflowComponent).(string)
				return in, nil
			},
			SuccessStatusCode: fasthttp.StatusAccepted,
		})
}
```

Then just pass the handler in the router, for example:

```go
		{
			Methods: []string{fasthttp.MethodPost},
			Route:   "workflows/{workflowComponent}/{instanceID}/pause",
			Version: apiVersionV1alpha1,
			Handler: a.onWorkflowActivityHandler(a.universal.PauseWorkflowAlpha1),
		},
```

Reply:
Sure, done